nautobot/extras/api/serializers.py

```diff
@@ -820,6 +822,50 @@ class Meta:
         ]
 
 
+#
+# Job Hooks
+#
+
+
+class JobHookSerializer(ValidatedModelSerializer):
+    url = serializers.HyperlinkedIdentityField(view_name="extras-api:jobhook-detail")
+    content_types = ContentTypeField(
+        queryset=ContentType.objects.filter(FeatureQuery("webhooks").get_query()).order_by("app_label", "model"),
+        many=True,
+    )
+
+    class Meta:
+        model = JobHook
+        fields = [
+            "id",
+            "url",
+            "name",
+            "content_types",
+            "job",
+            "enabled",
+            "type_create",
+            "type_update",
+            "type_delete",
+        ]
+
+    def validate(self, data):
+        validated_data = super().validate(data)
+
+        conflicts = JobHook.check_for_conflicts(
+            instance=self.instance,
+            content_types=data.get("content_types"),
+            job=data.get("job"),
+            type_create=data.get("type_create"),
+            type_update=data.get("type_update"),
+            type_delete=data.get("type_delete"),
+        )
+
+        if conflicts:
+            raise serializers.ValidationError(conflicts)
```

Comment:
This can go away with this logic moved to the model's `clean()` method! :)

Reply:
Looks like it was done this way because we need to validate content_types which is a ManyToManyField and we can't validate it in the clean method

Reply:
Okie dokie.